src/lib/content/coordinators/GetHierarchy.ts

```diff
@@ -0,0 +1,59 @@
+import { HierarchyContentItem, HierarchyRequest } from '../model/ByHierachy';
+import { ContentItem } from '../model/ContentItem';
+import { ContentBody } from '../model/ContentBody';
+
+export class HierarchyURLBuilder {
+  static HIERARCHY_URL = '/content/hierarchies/descendants/id/';
+  static MAXIMUM_DEPTH_PARAM = 'hierarchyDepth';
+  static MAXIMUM_PAGE_SIZE_PARAM = 'maxPageSize';
+  static LAST_EVALUATED_PARAM = 'pageCursor';
+
+  buildUrl(request: HierarchyRequest): string {
+    let url = HierarchyURLBuilder.HIERARCHY_URL;
+    url += request.rootId;
```

Comment:
I would use the `new URL` to contruct these searchParams as to avoid edgecasey bugs.

Example:
```js
const url = new URL(request.rootId, HierarchyURLBuilder.HIERARCHY_URL);

if (request.maximumDepth) {
  url.searchParams.append(HierarchyURLBuilder.MAXIMUM_DEPTH_PARAM, request.maximumDepth);
}

if (request.maximumPageSize) {
  url.searchParams.append(HierarchyURLBuilder.MAXIMUM_PAGE_SIZE_PARAM, request.maximumPageSize);
}

if (request.pageCursor) {
  url.searchParams.append(HierarchyURLBuilder.LAST_EVALUATED_PARAM, request.pageCursor);
}

return url.toString();
```